relay-server/src/actors/processor.rs

```diff
@@ -2741,70 +2741,80 @@ impl EnvelopeProcessorService {
         }
     }
 
-    fn handle_process_metrics(&self, message: ProcessMetrics) {
-        let ProcessMetrics {
-            items,
-            project_key: public_key,
-            start_time,
-            sent_at,
-        } = message;
-
-        let received = relay_common::time::instant_to_date_time(start_time);
-        let received_timestamp = UnixTimestamp::from_secs(received.timestamp() as u64);
-
-        let clock_drift_processor =
-            ClockDriftProcessor::new(sent_at, received).at_least(MINIMUM_CLOCK_DRIFT);
-
-        for item in items {
-            let payload = item.payload();
-            if item.ty() == &ItemType::Statsd {
-                let mut buckets = Vec::new();
-                for bucket_result in Bucket::parse_all(&payload, received_timestamp) {
-                    match bucket_result {
-                        Ok(mut bucket) => {
-                            clock_drift_processor.process_timestamp(&mut bucket.timestamp);
-                            buckets.push(bucket);
-                        }
-                        Err(error) => relay_log::debug!(
+    fn extract_metric_buckets_from_item(
```

Comment:
Looking for recommendation of where to put fn extract_metric_buckets_from_item. I know it's not good to leave it as an associated function. Would perhaps a method on item itself be fitting? I thought maybe methods on our fundamental types shouldn't have a lot of logic, and no logs. But may be wrong.

Reply:
I wish I could use IntoMetric trait but as the name suggests, it's for a singular Metric (now bucket).